perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/MembersManagerBlImpl.java

```diff
@@ -2201,22 +2201,41 @@ public void sendPasswordResetLinkEmail(PerunSession sess, Member member, String
 		for (Attribute a : logins) {
 			if (a.getFriendlyNameParameter().equals(namespace)) found = true;
 		}
-		if (!found) throw new InternalErrorException(user.toString()+" doesn't have login in namespace: "+namespace);
+		if (!found)
+			throw new InternalErrorException(user.toString() + " doesn't have login in namespace: " + namespace);
 
 		String email = "";
 		try {
-			Attribute a = perunBl.getAttributesManagerBl().getAttribute(sess, user, AttributesManager.NS_USER_ATTR_DEF+":preferredMail");
+			Attribute a = perunBl.getAttributesManagerBl().getAttribute(sess, user, mailAttributeUrn);
```

Comment:
The attribute does not have to be associated only with User entity. E.g. if you have urn `urn:perun:member:attribute-def:def:mail` this will fail because of `WrongAttributeAssignmentException`.  You have to check if the urn is for Member or User and then call `getAttribute` with either member or user.